[merged] Valgrind update

cgwalters opened the pull request:
Presumably the valgrind tests hadn't been working for anyone, but in their existing form they could only work on e.g. Fedora and not RHEL 7 as I'm using as they hardcoded a libc version in the supression file, etc.

hughsie commented:
looks good to me :+1:

cgwalters commented:
@hughsie - say: `@` `cgwalters-bot r+ 2b08b41` with no space between `@` and the name.

jsilhan commented:
Great.

> in their existing form they could only work on e.g. Fedora and not RHEL 7

@cgwalters, can you enable the tests for Fedora in spec then? I.e. discard `-DDISABLE_VALGRIND=1` for python2 and python3 builds, please. Then Jenkins will check pull requests with valgrind tests too (when he is ready).

cgwalters commented:
Rebased and updated.  I was getting a really strange issue where `make test` worked but `tito build --test --rpm` had different valgrind errors.  Couldn't figure out the environmental difference, ended up just adding another suppression (which I need to upstream in cockpit).

jsilhan commented:
```
INFO Results of the build can be found at: https://copr-be.cloud.fedoraproject.org/results/rpmsoftwaremanagement/libhif-pull-requests/fedora-23-i386/00169200-libhif (FAILED), https://copr-be.cloud.fedoraproject.org/results/rpmsoftwaremanagement/libhif-pull-requests/fedora-24-x86_64/00169200-libhif (FAILED), https://copr-be.cloud.fedoraproject.org/results/rpmsoftwaremanagement/libhif-pull-requests/fedora-24-i386/00169200-libhif (FAILED), https://copr-be.cloud.fedoraproject.org/results/rpmsoftwaremanagement/libhif-pull-requests/fedora-23-x86_64/00169200-libhif (FAILED), https://copr-be.cloud.fedoraproject.org/results/rpmsoftwaremanagement/libhif-pull-requests/fedora-rawhide-x86_64/00169200-libhif (RUNNING), https://copr-be.cloud.fedoraproject.org/results/rpmsoftwaremanagement/libhif-pull-requests/fedora-rawhide-i386/00169200-libhif (RUNNING)
```

jsilhan commented:
both libhif and hawkey valgrid checks fail. In the spec it's build insource from cmake so that might be the issue as it could not find the proper dir with suppressed warnings. If `tito build --rpm --test` passes locally then Jenkins test shoudl be alright as well.

cgwalters commented:
Okay, I think there are a few things going on here:

 - Without -debuginfo installed, valgrind may not be able to properly match up the suppressions.  Which means to valgrind in copr (or koji) we'd need to `BuildRequires: glib2-debuginfo rpm-debuginfo ...`.  Is there any precedent for that even?  Pretty ugly...
 - For some reason on f23 there are lua warnings not in at least rhel7.  And this is the real problem with automated valgrind is that basically it becomes a long-term maintenance burden as you have to track suppressions for all of your dependencies.

cgwalters commented:
So...we could back out enabling it by default and make it a separate `make test-valgrind`?

cgwalters commented:
I go back and forth on this...having the valgrind tests obviously found real issues, but I'm struggling to justify the time for generating supressions and auditing all of the stack across multiple major releases as shared libraries change underneath us =(

cgwalters commented:
We'll make this a separate makefile target for now.

jsilhan commented:
actually we can have it as it is and merge this PR without b310ce1. Valgrind tests should pass on F23 with debuginfo packages currently. We can add new variable `valgrind` into spec and if set, it will specify debuginfo buildrequires and will run valgrind checks - which will be for F23 only in Jenkins CI. Maintaining valgrind suppression warnings up-to-date for one latest stable fedora release could be done with little effort. Sounds reasonable?

cgwalters commented:
Okay, replaced the last commit :arrow_up:

jsilhan commented:
@cgwalters-bot r+ 9d0cacf
